kernel/aster-nix/src/fs/utils/page_cache.rs

```diff
@@ -66,6 +67,51 @@ impl PageCache {
     pub fn backend(&self) -> Arc<dyn PageCacheBackend> {
         self.manager.backend()
     }
+
+    /// Resizes the current page cache to a target size.
+    pub fn resize(&self, new_size: usize) -> Result<()> {
+        // If the new size is smaller and not page-aligned,
+        // first zero the gap between the new size and the
+        // next page boundry (or the old size), if such a gap exists.
+        let old_size = self.pages.size();
+        if old_size > new_size && new_size % PAGE_SIZE != 0 {
+            let gap_size = old_size.min(new_size.align_up(PAGE_SIZE)) - new_size;
+            if gap_size > 0 {
+                self.zero_range(new_size..new_size + gap_size)?;
+            }
+        }
+        self.pages.resize(new_size)
+    }
+
+    /// Zeroing a range of cached data.
+    pub fn zero_range(&self, range: Range<usize>) -> Result<()> {
+        if range.is_empty() {
+            return Ok(());
+        }
+        let (start, end) = (range.start, range.end);
+        let zeroed_page_buf = vec![0u8; PAGE_SIZE];
```

Comment:
I found a little gap here since
1. `VmIo` doesn't provide `fill()` for now, only `write_bytes()`/`write_slice()`/`write_val()` which all end to `write_bytes(..., buf: &[u8])` that takes a u8 slice;
2. `fill()` is provided in `VmWriter` now, but for now there is no way for `Vmo` of `PageCache` to create a `VmWriter`, `Frame::writer()` will do but also no way to take `Frame` of `Vmo`.